### PR TITLE
Open up access to CharacterHit for subclasses of StyledTextArea

### PR DESCRIPTION
**Pull request by JordanMartinez**

### CODE REVIEW

**Comment by TomasMikula** [comment]
Hmm, I haven't noticed that `ParagraphBox` became public. I don't think it should be. Then only the `hit(double x, double y)` method can be made public.

**Comment by JordanMartinez** [comment]
It's public even before the skin removal process. Should I make both the class and its constructor package private in this branch or via a new one?

**Comment by JordanMartinez** [comment]
It's changed to public in this commit: 0ae84c25936bdb285434ecc96564f0b1f946221d

**Comment by TomasMikula** [comment]
Either way ;)

**Comment by JordanMartinez** [comment]
`ParagraphBox`'s constructor is public long before this mistake. I'm guessing it's public so it could be accessed by the main richtextfx package outside the skin package?

However, now that it's inside the richtextfx package, it doesn't need to be public anymore.

File: richtextfx/src/main/java/org/fxmisc/richtext/StyledTextArea.java [hunk 1]
```diff
@@ -677,7 +677,7 @@ CharacterHit hit(ParagraphBox.CaretOffsetX x, double y) {
         }
     }
 
-    CharacterHit hit(double x, double y) {
+    protected CharacterHit hit(double x, double y) {
```
Comment (TomasMikula) [inline richtextfx/src/main/java/org/fxmisc/richtext/StyledTextArea.java]: Don't we want this one public and keep the other two package private?

Comment (JordanMartinez) [inline richtextfx/src/main/java/org/fxmisc/richtext/StyledTextArea.java]: I guess that would be better overall....

**Comment by JordanMartinez** [comment]
I added javadoc that gives an example of what it can do, too.

**Comment by TomasMikula** [comment]
:+1:

**Pull request merged**